src/util.js

```diff
@@ -171,7 +171,19 @@ export function createRegExpRestore () {
     // Create the regular expression that will reconstruct the RegExp properties
     ret.exp = new RegExp(arrJoin.call(reg, '') + lm, ml);
 
-    return ret;
+    return function() {
+      try {
+        ret.exp.test(ret.input);
+      } catch(e) {
+        // Warn instead of fail if RegExp was too big
+        if (e && e.message && e.message.slice(-14) === 'RegExp too big') {
+          console && console.warn && console.warn('Error restoring RegExp properties. RegExp too big.');
```

Comment:
This code is specifically so it doesn't throw an error with as much detail as possible. Our projects dependency's dependency is running a regex on a multi megabyte string. That then gets written to the console when the regex property restorer fails. 

This is bad for a couple of reasons. 1) our site breaks because the polyfill is trying to restore properties that nobody cares about. 2) the code running the regex isn't code under our control. 3) there's a multi megabyte string being written to the console and causes bad browser behaviour. 4) the regex that is output in the error is based on the input to the regexp not the regexp itself, so makes a poor approximation that isn't easily searched in  the code.

I can update the warning to include the first n chars of the approximated regex if that's an acceptable compromise?

The above code makes Intl do a best-effort restore of the regexp properties while not bombing out on error. 

The native implementation doesn't error depending on prior regex run and I don't think the polyfill should either.

If no change is made in Intl.js then we would need to run a dummy regexp before every Intl method to make sure the code doesn't error. That doesn't seem like a reasonable expectation for use of this polyfill.